src/com/amazon/corretto/crypto/provider/RsaCipher.java

```diff
@@ -321,8 +391,13 @@ protected void engineSetMode(final String mode) throws NoSuchAlgorithmException
     }
 
     @Override
-    protected void engineSetPadding(final String padding) throws NoSuchPaddingException {
-        if (!padding_.jceName.equalsIgnoreCase(padding)) {
+    protected void engineSetPadding(final String newPadding) throws NoSuchPaddingException {
+        if (newPadding == Padding.OAEP.paddingStr) {
+            synchronized(lock_) {
+                oaepParams_ = OAEPParameterSpec.DEFAULT;
+                paddingSize_ = calculateOaepPaddingLen(OAEPParameterSpec.DEFAULT.getDigestAlgorithm());
+            }
+        } else if (!padding_.paddingStr.equalsIgnoreCase(newPadding)) {
```

Comment:
i haven't checked the coverage here, but i believe we do require this if we want to remain backwards compat with `RSA/ECB/OAEPWithSHA-1AndMGF1Padding` as Cipher [calls this method](https://github.com/corretto/corretto-11/blob/14c02261590b4dc01284888a7a51d39ff581ac8d/src/java.base/share/classes/javax/crypto/Cipher.java#L391) with the padding configuration.

Reply:
I don't think it's actually called. I think we'd see this path if we advertised `RSA` and the various padding modes separately rather than together.

The SunJCE provider advertises them separately:
```
$ java JavaModeDump | grep 'Cipher.RSA'
Cipher.RSA SupportedKeyClasses = java.security.interfaces.RSAPublicKey|java.security.interfaces.RSAPrivateKey
Cipher.RSA SupportedModes = ECB
Cipher.RSA = com.sun.crypto.provider.RSACipher
Cipher.RSA SupportedPaddings = NOPADDING|PKCS1PADDING|OAEPPADDING|OAEPWITHMD5ANDMGF1PADDING|OAEPWITHSHA1ANDMGF1PADDING|OAEPWITHSHA-1ANDMGF1PADDING|OAEPWITHSHA-224ANDMGF1PADDING|OAEPWITHSHA-256ANDMGF1PADDING|OAEPWITHSHA-384ANDMGF1PADDING|OAEPWITHSHA-512ANDMGF1PADDING|OAEPWITHSHA-512/224ANDMGF1PADDING|OAEPWITHSHA-512/256ANDMGF1PADDING
```

If we were to do the same with the current version of ACCP (which doesn't have the extended OAEP support) you'll see the following:
```
$ java JavaModeDump | grep 'Cipher.RSA'
Cipher.RSA/ECB/OAEPWithSHA-1AndMGF1Padding = com.amazon.corretto.crypto.provider.RsaCipher$OAEPSha1
Cipher.RSA/ECB/NoPadding = com.amazon.corretto.crypto.provider.RsaCipher$NoPadding
Cipher.RSA/ECB/Pkcs1Padding = com.amazon.corretto.crypto.provider.RsaCipher$Pkcs1
```

Reply:
<img width="895" alt="Screen Shot 2022-05-19 at 11 11 22 AM" src="https://user-images.githubusercontent.com/3589880/169330489-7dac7cca-d954-4baf-855a-50e168d0c831.png">

huh, yeah you were right. we don't call this. I'll remove it.